### PR TITLE
feat(cce): add some new parameters in the cluster

### PR DESCRIPTION
**Pull request by Jason-Zhang9309**
**What this PR does / why we need it**:

**Which issue this PR fixes**:
*(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close that issue when PR gets merged)*
fixes #xxx

**Special notes for your reviewer**:

**Release note**:

```release-note
1. add new parameters alias, custom_san, ipv6_enable, support_istio and component_congurations
2. add new attribute category
3. make container_network_cidr updatable
```

## PR Checklist

* [x] Tests added/passed.
* [x] Documentation updated.
* [x] Schema updated.

## Acceptance Steps Performed

```
make testacc TEST='./huaweicloud/services/acceptance/cce' TESTARGS='-run TestAccCluster_basic'
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./huaweicloud/services/acceptance/cce -v -run TestAccCluster_basic -timeout 360m -parallel 4
=== RUN   TestAccCluster_basic
=== PAUSE TestAccCluster_basic
=== CONT  TestAccCluster_basic
--- PASS: TestAccCluster_basic (591.30s)
PASS
ok      github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/services/acceptance/cce       591.356s

make testacc TEST='./huaweicloud/services/acceptance/cce' TESTARGS='-run TestAccCluster_multiContainerNetworkCidrs'
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./huaweicloud/services/acceptance/cce -v -run TestAccCluster_multiContainerNetworkCidrs -timeout 360m -parallel 4
=== RUN   TestAccCluster_multiContainerNetworkCidrs
=== PAUSE TestAccCluster_multiContainerNetworkCidrs
=== CONT  TestAccCluster_multiContainerNetworkCidrs
--- PASS: TestAccCluster_multiContainerNetworkCidrs (529.56s)
PASS
ok      github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/services/acceptance/cce       529.610s
```

### CODE REVIEW

File: docs/resources/cce_cluster.md [hunk 1]
```diff
@@ -343,6 +358,16 @@ The `extend_params` block supports:
   Defaults to none.  
   Changing this parameter will create a new cluster resource.
 
+<a name="cce_cluster_component_congurations"></a>
+The `omponent_congurations` block supports:
```
Comment (ShiChangkuo) [inline docs/resources/cce_cluster.md]:
```suggestion
The `component_congurations` block supports:
```

File: docs/resources/cce_cluster.md [hunk 2]
```diff
@@ -343,6 +358,16 @@ The `extend_params` block supports:
...
+
+* `name` - (Optional, String, ForceNew) Specifies the component name.
+  This parameter is mandatory for dedicated CCE clusters.
+
+  Changing this parameter will create a new cluster resource.
+
+* `congurations` - (Optional, String, ForceNew) Specifies json string the component configurations.
```
Comment (ShiChangkuo) [inline docs/resources/cce_cluster.md]:
```suggestion
* `congurations` - (Optional, String, ForceNew) Specifies JSON string of the component configurations.
```

File: huaweicloud/services/cce/resource_huaweicloud_cce_cluster.go [hunk 3]
```diff
@@ -292,6 +298,43 @@ func ResourceCluster() *schema.Resource {
 				Type:     schema.TypeBool,
 				Optional: true,
 			},
+			"component_congurations": {
+				Type:     schema.TypeList,
+				Optional: true,
+				ForceNew: true,
+				Elem: &schema.Resource{
+					Schema: map[string]*schema.Schema{
```
Comment (ShiChangkuo) [inline huaweicloud/services/cce/resource_huaweicloud_cce_cluster.go:306]: `name` and `congurations` should be required?

File: docs/resources/cce_cluster.md (hunk 1 above)
Comment (Jason-Zhang9309) [reply docs/resources/cce_cluster.md]: fixed

File: docs/resources/cce_cluster.md (hunk 2 above)
Comment (Jason-Zhang9309) [reply docs/resources/cce_cluster.md]: fixed

File: huaweicloud/services/cce/resource_huaweicloud_cce_cluster.go (hunk 3 above)
Comment (Jason-Zhang9309) [reply huaweicloud/services/cce/resource_huaweicloud_cce_cluster.go:306]: make the `name` required

**Pull request merged**